### PR TITLE
fix links to docs

### PR DESCRIPTION
**Pull request by WestleyArgentum**
These links weren't working for me. A couple returned 404 and the one about debugging segfaults said "permission denied" (how rude...)

It's possible these links are changing moving forward? Not sure, but this made them work for me for now.

### CODE REVIEW

**Review by yuyichao — CHANGES_REQUESTED** [review]
Please don't link to stable doc. The new doc is at  https://docs.julialang.org/en/latest

**Comment by StefanKarpinski** [comment]
So the correct fix here is to link to latest docs hosted at docs.julialang.org and not include trailing slashes in URL names, e.g. `https://docs.julialang.org/en/latest/manual/packages`. Any doc links to http://julia.readthedocs.io/ should be changed and any trailing slashes should be stripped.

**Comment by WestleyArgentum** [comment]
updated and ran threw all the links to make them consistent, let me know if that looks better

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
 - Contributing code? Be sure to review the [contributor checklist](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#contributor-checklist) for helpful tips on the tools we use to build Julia.
 
-- Library feature requests are generally not accepted on this issue tracker. New libraries should be developed as [packages](https://docs.julialang.org/en/release-0.4/manual/packages/#package-development). Discuss ideas for libraries at the [Julia Discourse forum](https://discourse.julialang.org/). Doing so will often lead to pointers to existing projects and bring together collaborators with common interests.
+- Library feature requests are generally not accepted on this issue tracker. New libraries should be developed as [packages](https://docs.julialang.org/en/release-0.4/manual/packages/#package-development). Discuss ideas for libraries at the [Julia Discourse forum](https://discourse.julialang.org). Doing so will often lead to pointers to existing projects and bring together collaborators with common interests.
```
Comment (Sacha0) [inline CONTRIBUTING.md]: Should the link to the packages documentation point to latest rather than release-0.4?

Comment (WestleyArgentum) [inline CONTRIBUTING.md]: oh, good catch

**Review by Sacha0 — APPROVED** [review]
Thanks @WestleyArgentum! :) (I have not tested the links.)

**Pull request merged**